query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java

```diff
@@ -134,7 +129,7 @@ public boolean isRunning() {
             IndexWorker indexWork = new IndexWorker(cache.getName(), indexedTypes, skipIndex, null);
             futures.add(executor.submitConsumer(indexWork, TRI_CONSUMER));
             CompletableFuture<Void> compositeFuture = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
-            return compositeFuture.whenCompleteAsync(flushIfNeeded, localExecutor);
+            return compositeFuture.whenComplete(flushIfNeeded);
```

Comment:
Does `flushIfNeeded` not block?

Reply:
yes

Reply:
This change was not necessary either, I will rollback

Reply:
I always enjoy removing usages of blocking executor if we can :)